kagomegen.py

```diff
@@ -0,0 +1,188 @@
+    for i in range(len(site_pos)):
+        ind = []
+        for j in range(len(site_pos[i][0])):
+            # finding the index of the neighbour sites in sites[]
+            us = list(np.where(np.round(sites, 5) ==
+                               np.round(site_pos[i][0][j], 5)))[0]
+            # print(us)
+            res = [idx for idx, val in enumerate(us) if val in us[:idx]]
+            # print(res)
+            if res != []:
+                ind.append(us[res[0]])
+
+        if periodic == 1:
+            if len(ind) != 4:
+                if i in boundary_ind_b:
+                    if i < (Lx*3*(Ly-1)+1):
+                        ind.append(boundary_ind_c[Ly+v1])
+                        v1 = v1+1
+                    if i >= (Lx*3*(Ly-1)+1):
+                        ind.append(boundary_ind_a[Lx+v2])
+                        ind.append(boundary_ind_c[Ly+v1])
+                        v2 = v2+1
+                elif i in boundary_ind_a:
+                    if i <= (Ly*3-1):
+                        ind.append(boundary_ind_b[Ly+v3])
+                        v3 = v3+1
+                    if (i > (Ly*3-1) or i == 0):
+                        ind.append(boundary_ind_c[Lx+v4])
+                        v4 = v4+1
+                else:
+                    if i < (Ly*3-1):
+                        ind.append(boundary_ind_b[Ly+1+v5])
+                        v5 = v5+1
+                    else:
+                        ind.append(boundary_ind_a[v6])
+                        ind.append(boundary_ind_b[v6])
+                        v6 = v6+1
+            ind.sort()
+            # if len(ind) !=4:
+            #     print('bad')
+            # else:
+            #     print(i,ind)
+
+        for k in ind:
+            if [i, k] and [k, i] not in adjacency_list:  # removing symmetery
+                adjacency_list.append([i, k])
+
+    adjacency_list = np.array(adjacency_list)
+    return adjacency_list
+
+
+def make_kagome(Lx, Ly, periodic, name, comment):
+
+    lat = Lattice(name, comment)
+    size = Lx*Ly
+    num_sites = size*3
+
+    # translation vector
+    trans = [1.0, np.sqrt(3)]
+    a, b, c, sites = position(Lx, Ly, trans)
+    site_pos = nearest_neighbours(size, a, b, c)
+    # boundary sites
+    boundary_sites_a = sites[0::Lx*3]+sites[0:Ly*3:3]
+    boundary_sites_b = sites[1::Lx*3]+sites[-(Ly*3-1)::3]
+    boundary_sites_c = sites[2:(Ly*3):3]+sites[(Ly*3)-1::(Lx*3)]
+
+    boundary_ind_a = _finding_index(boundary_sites_a)
+    boundary_ind_b = _finding_index(boundary_sites_b)
+    boundary_ind_c = _finding_index(boundary_sites_c)
+
+    adjacency = adjacencyList(Lx, Ly,
+                              site_pos, sites, boundary_ind_a, boundary_ind_b, boundary_ind_c, periodic)
+    # hopping matrix of equal amplitude
+    hopping_matrix = np.ones(len(adjacency))
+    positions = np.zeros((num_sites, 3))
+    positions[:, :-1] = sites  # making 3-dim
+
+    lat.sites.append(Site(adjacency, positions, hopping_matrix))
```

Comment:
You need to append each site separately. At the moment, this produces a lattice with only one site whose index is the adjacency list and its position is the list of all positions.
See https://github.com/chelseajohn/latgraph/blob/e9612f2f35c174038db9360481e3ba6730660eca/lattice.py#L50 for the constructor arguments for `Site`.

Reply:
okay. Now how to do I pass adjacency_list to lattice then?

Reply:
You need to loop over the lattice sites and for each create a `Site` object:
```.py
lattice.sites.append(Site(index, pos, neighbours, hoppings))
```
where
- `index`: integer, your sites need to be labeled by an index from 0 to N
- `pos`: 3D position as numpy array
- `neighbours`: tuple or list of indices of all neighbours of that site
- `hoppings`: tuple or list of hoppings strengths, can just be `[1]*len(neighbours)`

Reply:
I got that, I was asking if there was another function that would create the adjacency matrix from the data given

Reply:
Nevermind. Everything works now